addons/binding/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandler.java

```diff
@@ -0,0 +1,673 @@
+                                    if (i++ == 0) {
+                                        flUpdatePower = true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            } else if (jsonEvent.contains("new_play_mode")) {
+                Integer playMode = getJInt(event, "new_play_mode", -1);
+                Integer duration = getJInt(event, "duration", -1);
+                Integer playPostion = getJInt(event, "playPostion", -1);
+                logger.debug("STB event playContent: playMode={}, duration={}, playPosition={}",
+                        control.getPlayStatus(playMode), duration.toString(), playPostion.toString());
+
+                // If we get a playConfig event there MR must be online. However it also sends a
+                // plyMode stop before powering off the device, so we filter this.
+                if ((playMode != EV_PLAYCHG_STOP) && this.isInitialized()) {
+                    flUpdatePower = true;
+                }
+
+                if (playMode != -1) {
+                    updateState(CHANNEL_PLAY_MODE, new StringType(control.getPlayStatus(playMode)));
+                }
+                if (duration > 0) {
+                    updateState(CHANNEL_PROG_DURATION, new StringType(duration.toString()));
+                }
+                if (playPostion != -1) {
+                    updateState(CHANNEL_PROG_POS, new StringType(Integer.toString(playPostion)));
+                }
+            } else {
+                logger.debug("Unknown stb event, JSON='{}", jsonEvent);
+            }
+            if (flUpdatePower) {
+                // We received a non-stopped event -> MR must be on
+                updateState(CHANNEL_POWER, OnOffType.ON);
+            }
+        } catch (Exception e) {
+            logger.error("Unable to process STB event: {}, JSON='{}'", e.getMessage(), jsonEvent);
+        } finally {
+            if (reader != null) {
+                reader.close();
+            }
+        }
+    }
+
+    @Override
+    public void onPowerOff() throws Exception {
+        logger.info("Power-Off received for device '{}' ({})", thingConfig.getFriendlyName(),
+                thingConfig.getTerminalID());
+        // MR was powered off -> update pwoer status, reset items
+        updateState(CHANNEL_POWER, OnOffType.OFF);
+        updateState(CHANNEL_PROG_DURATION, new StringType(""));
+        updateState(CHANNEL_PROG_POS, new StringType(""));
+        updateState(CHANNEL_PROG_TITLE, new StringType(""));
+        updateState(CHANNEL_PROG_TEXT, new StringType(""));
+        updateState(CHANNEL_PROG_START, new StringType(""));
+        updateState(CHANNEL_PROG_DURATION, new StringType(""));
+    }
+
+    @Override
+    public void dispose() {
+        scheduler.shutdownNow();
+        cancelPairingCheck();
```

Comment:
First cancel jobs, then shutdown a scheduler. If this is the frameworks scheduler -> don't shut it down at all. My god. Bad things will happen.

Reply:
scheduler is a private object in the class:
    private ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);

both operation do apply for independing jobs/objects, however I swapped the lines